src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/RecreateAppealPdfHandler.java

```diff
@@ -1,65 +1,75 @@
+import uk.gov.hmcts.reform.sscs.ccd.domain.*;
 import uk.gov.hmcts.reform.sscs.idam.IdamService;
 import uk.gov.hmcts.reform.sscs.idam.IdamTokens;
+import uk.gov.hmcts.reform.sscs.service.EmailService;
+import uk.gov.hmcts.reform.sscs.service.SscsPdfService;
 
-@Service
+@Component
 @Slf4j
-public class EventService {
-
-    private static final Logger LOG = getLogger(EventService.class);
+public class RecreateAppealPdfHandler implements PreSubmitCallbackHandler<SscsCaseData> {
 
     private final SscsPdfService sscsPdfService;
     private final EmailService emailService;
     private final IdamService idamService;
 
     @Autowired
-    EventService(SscsPdfService sscsPdfService,
-                 IdamService idamService,
-                 EmailService emailService) {
+    public RecreateAppealPdfHandler(SscsPdfService sscsPdfService, EmailService emailService, IdamService idamService) {
         this.sscsPdfService = sscsPdfService;
-        this.idamService = idamService;
         this.emailService = emailService;
+        this.idamService = idamService;
     }
 
-    public boolean handleEvent(EventType eventType, SscsCaseData caseData) {
+    @Override
+    public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        boolean canHandle = callbackType == CallbackType.ABOUT_TO_SUBMIT
+                && callback.getEvent() == EventType.CREATE_APPEAL_PDF;
+        CaseDetails<SscsCaseData> caseDetails = callback.getCaseDetails();
+        log.info("Can handle callback for case [" + caseDetails.getCaseData().getCcdCaseId() + "]"
+                + " for type [" + callbackType + "]"
+                + " event [" + callback.getEvent() + "]"
+                + " can handle [" + canHandle + "]");
+        return canHandle;
+    }
 
-        if (CREATE_APPEAL_PDF == eventType) {
-            createAppealPdfAndSendToRobotics(caseData);
-            return true;
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        if (!canHandle(callbackType, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
         }
 
-        return false;
-    }
+        CaseDetails<SscsCaseData> caseDetails = callback.getCaseDetails();
+        SscsCaseData caseData = caseDetails.getCaseData();
+        log.info("Handling create appeal pdf event for case [" + caseData.getCcdCaseId() + "]");
+
+        PreSubmitCallbackResponse<SscsCaseData> sscsCaseDataPreSubmitCallbackResponse = new PreSubmitCallbackResponse<>(caseData);
+
+        createAppealPdfAndSendToRobotics(caseData);
 
-    private void handleEvent(Runnable eventHandler) {
-        Executors.newSingleThreadExecutor().submit(eventHandler);
+        return sscsCaseDataPreSubmitCallbackResponse;
     }
 
+
     private void createAppealPdfAndSendToRobotics(SscsCaseData caseData) {
+        //FIXME: This code should be refactored to use the PDF generation for SYA
```

Comment:
should this be fixed now?

Reply:
Na, i've put it in there as a reminder that this code is rubbish and needs to be fixed at some point